Shared/Extensions/URLExtensions.swift

```diff
@@ -320,7 +322,8 @@ extension URL {
     :returns: The public suffix for within the given hostname.
     */
   public var publicSuffix: String? {
-    return host.flatMap { publicSuffixFromHost($0, withAdditionalParts: 0) }
+    let registry = (self as NSURL).registry
```

Comment:
Can we add a helper method for `URL` extension so we don't have to bridge it to `NSURL` every time?